src/Gamepad.js

```diff
@@ -0,0 +1,78 @@
+let gamepadId = 0;
+
+import { GamepadButton } from "./GamepadButton.js";
+
+export class Gamepad extends HTMLDivElement {
+  constructor() {
+    super();
+
+    const id = gamepadId++;
+
+    // setup the gamepad's front end
+    this.innerHTML = `
+      <style>
+        .gamepad {
+          border: 1px solid grey;
+          display: inline-block;
+          margin: 5px;
+        }
+      </style>
+      Hello gamepad ${id}!
+    `;
+    this.classList.add("gamepad");
+
+    // set default attributes
+    this._id = id;
+    this._index = 0; // temporary value for now
+    this._connected = false;
+    this._timestamp = new Date().getTime();
```

Comment:
You can get a DOMHighRestTimestamp from https://www.w3.org/TR/hr-time-2/#sec-performance
 
```suggestion
    this._timestamp = performance.now();
```

Remember, if you see a weird IDL type, it's usually defined somewhere how you get one.